### PR TITLE
Introduce `wp_cache_supports()`

### PR DESCRIPTION
**Pull request by SergeyBiryukov**
Trac ticket: https://core.trac.wordpress.org/ticket/56605

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by JJJ — APPROVED** [review]
This looks good to me (as part 1 of 2) 👍

Part 2: https://core.trac.wordpress.org/ticket/56605?replyto=7#comment:5

**Comment by tillkruss** [comment]
Are there any other features like the `wp_cache_*_multiple()` methods we want to allow checking support for? It would be handy to have for various diagnostics.

File: src/wp-includes/cache.php [hunk 1]
```diff
@@ -302,15 +302,15 @@ function wp_cache_flush_group( $group ) {
...
- * @see WP_Object_Cache::flush_group()
- *
- * @return bool True if group flushing is supported, false otherwise.
+ * @param string $feature Name of the feature to check for.
+ *                        Possible values include: 'flush_group'.
+ * @return bool True if the feature is supported, false otherwise.
  */
-function wp_cache_supports_group_flush() {
+function wp_cache_supports( $feature ) {
 	return true;
```
Comment (tillkruss) [inline src/wp-includes/cache.php]: Maybe we add the framework already?

```suggestion
	switch ( $feature ) {
		'flush_group':
		    return true;
		default:
		    return false;
	}
```

Comment (spacedmonkey) [reply src/wp-includes/cache.php]: I wonder if we should add, get_multiple, delete_mulitple, update_multiple and flush_runtime

Comment (tillkruss) [reply src/wp-includes/cache.php]: Yeah, exactly 👍

**Review by spacedmonkey — APPROVED** [review]

File: src/wp-includes/cache.php (hunk 1 above)
Comment (SergeyBiryukov) [reply src/wp-includes/cache.php]: Added the `switch` statement. :+1:

@tillkruss @spacedmonkey @JJJ If we add the other features here, do we need to make any changes to the functions in `wp-includes/cache-compat.php` to check for `wp_cache_supports()`? 

It looks like these functions are polyfilled anyway, so we could add these features in `wp_cache_supports()`, but the functions themselves would not need any changes:
* `wp_cache_add_multiple()`
* `wp_cache_set_multiple()`
* `wp_cache_get_multiple()`
* `wp_cache_delete_multiple()`

`wp_cache_flush_runtime()`, however, may need an `if ( ! wp_cache_supports( 'flush_runtime' ) )` check similar to the one in `wp_cache_flush_group()`:
```
if ( ! wp_cache_supports( 'flush_runtime' ) ) {
	_doing_it_wrong(
		__FUNCTION__,
		__( 'Your object cache implementation does not support flushing the in-memory runtime cache.' ),
		'6.1.0'
	);

	return false;
}
```

Comment (tillkruss) [reply src/wp-includes/cache.php]: Yeah, I was mostly thinking of 3rd party plugins correctly listing their supported features.

**Comment by SergeyBiryukov** [comment]
Merged in [r54448](https://core.trac.wordpress.org/changeset/54448).

**Pull request closed without merging**